### PR TITLE
Format closed ranges

### PR DESCRIPTION
**Pull request by marcusklaas**
Closes https://github.com/rust-lang-nursery/rustfmt/issues/890.

### CODE REVIEW

File: tests/source/expr.rs [hunk 1]
```diff
@@ -244,3 +244,15 @@ fn issue767() {
...
+fn ranges() {
+    let x = aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa .. bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb;
+    let y = aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ... bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb;
+    let z = ... x ;
+    let infi_range_2 = ... ;
+
+    a ... b
+
+    // the expr below won't compile for some reason...
+    // let a = 0 ... ;
```
Comment (kamalmarhubi) [inline tests/source/expr.rs]: I commented on #893, but this isn't meant to compile, so the comment can be removed.

File: src/expr.rs [hunk 2]
```diff
@@ -176,48 +176,48 @@ impl Rewrite for ast::Expr {
...
             }
             // We do not format these expressions yet, but they should still
             // satisfy our width restrictions.
             ast::ExprKind::InPlace(..) |
-            ast::ExprKind::InlineAsm(..) => {
+            ast::ExprKind::InlineAsm(..) |
+            // TODO(#848): Handle type ascription
+            ast::ExprKind::Type(_, _) |
+            // TODO(#867): Handle try shorthand
+            ast::ExprKind::Try(_) => {
```
Comment (kamalmarhubi) [inline src/expr.rs]: is this the "default"?

Comment (nrc) [inline src/expr.rs]: Yes, it just uses the source code without formatting

Comment (kamalmarhubi) [inline src/expr.rs]: good to know for the future!

**Pull request merged**